UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.API/Services/SubscriptionService.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using FluentValidation.Results;
+using Microsoft.Extensions.Options;
+using UKHO.ExternalNotificationService.API.Validation;
+using UKHO.ExternalNotificationService.Common.Configuration;
+using UKHO.ExternalNotificationService.Common.Models.Request;
+
+namespace UKHO.ExternalNotificationService.API.Services
+{
+    public class SubscriptionService : ISubscriptionService
+    {
+        private readonly ID365PayloadValidator _d365PayloadValidator;
+        private readonly IOptions<D365PayloadKeyConfiguration> _d365PayloadKeyConfiguration;
+
+        public SubscriptionService(ID365PayloadValidator d365PayloadValidator,
+                                   IOptions<D365PayloadKeyConfiguration> d365PayloadKeyConfiguration)
+        {
+            _d365PayloadValidator = d365PayloadValidator;
+            _d365PayloadKeyConfiguration = d365PayloadKeyConfiguration;
+        }
+
+        public Task<ValidationResult> ValidateD365PayloadRequest(D365Payload d365Payload)
+        {
+            return _d365PayloadValidator.Validate(d365Payload);
+        }
+
+        public SubscriptionRequest ConvertToSubscriptionRequestModel(D365Payload d365Payload)
+        {
+            var inputParameter = d365Payload.InputParameters.Single();
+            var postEntityImage = d365Payload.PostEntityImages.SingleOrDefault(i => i.Key == _d365PayloadKeyConfiguration.Value.PostEntityImageKey);
```

Comment:
I agree with you, but right now we are getting an issue with update subscriptions because the PostEntityImages key name is “ENS Dev Payload” instead of “Subscription Image”.
The PostEntityImages key is being discussed in progress right now, all config changes will be removed once the discussion is completed.
We have made the key name configurable so that it can be changed for each environment.